pkg/metrics/metrics.go

```diff
@@ -0,0 +1,186 @@
+package metrics
+
+import (
+	"strconv"
+
+	"github.com/golang/glog"
+	mapiv1beta1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	machineinformers "github.com/openshift/cluster-api/pkg/client/informers_generated/externalversions/machine/v1beta1"
+	machinelisters "github.com/openshift/cluster-api/pkg/client/listers_generated/machine/v1beta1"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/utils/pointer"
+)
+
+var (
+	// MachineCountDesc is a metric about machine object count in the cluster
+	MachineCountDesc = prometheus.NewDesc("mapi_machine_items_total", "Count of machine objects currently at the apiserver", nil, nil)
+	// MachineSetCountDesc Count of machineset object count at the apiserver
+	MachineSetCountDesc = prometheus.NewDesc("mapi_machineset_items_total", "Count of machinesets at the apiserver.", nil, nil)
+
+	//MachineInfo Information of the Machines currently managed by the machine controller.
+	MachineInfo = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mapi_machine_info",
+		Help: "Information of the Machines currently managed by the machine controller.",
+	}, []string{"name", "namespace", "createdAt",
+		"spec_provider_id", "node", "api_version"})
+
+	// MachineSetInfo Information of the Machinesets currently reported by the apiserver
+	MachineSetInfo = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mapi_machine_set_info",
+		Help: "Information of the Machinesets currently managed by the machine controller",
+	}, []string{"name", "namespace", "createdAt", "api_version"})
+
+	// MachineSetInfoSpecReplicas Count of the Machinesets Spec Replicas.
+	MachineSetInfoSpecReplicas = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mapi_machine_set_info_spec_replicas",
+		Help: "Count of the Machinesets Spec Replicas.",
+	}, []string{"name", "namespace"})
+
+	// MachineSetStatusAvailableReplicas Information of the Machinesets' status for available replicas.
+	MachineSetStatusAvailableReplicas = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mapi_machine_set_status_available_replicas",
+		Help: "Information of the mapi managed Machinesets' status for available replicas.",
```

Comment:
Machinsets' <- typo?
Also here you say "mapi managed" whereus above " managed by the machine controller" can we use consistent message?